### PR TITLE
Avoid using pkg_resources

### PR DESCRIPTION
**Pull request by zsimic**
Don't import pkg_resources, unless absolutely necessary

### CODE REVIEW

File: src/runez/__main__.py [hunk 1]
```diff
@@ -224,53 +227,20 @@ def _show_fgcolors(bg=runez.plain, border=None):
...
-
-    except ImportError:
-        pass
-
-    import pkg_resources
-    import sysconfig
-
-    result = []
+def _interesting_top_levels():
+    uninteresting = re.compile(r"^(_|pip|pkg_resources|pydev|pytest|setuptools)")
```
Comment (thatch) [inline src/runez/__main__.py:231]: Is the end intentionally not anchored?

File: src/runez/__main__.py [hunk 2]
```diff
@@ -47,12 +49,13 @@ def cmd_diagnostics():
     """Show system diagnostics sample"""
     parser = runez.cli.parser()
     parser.add_argument("--border", default="colon", choices=NAMED_BORDERS, help="Use custom border.")
-    parser.add_argument("--pyenv", default="~/.pyenv", help="Pyenv folder to scan for python installations.")
+    parser.add_argument("--pyenv", default="PATH", help="Colon separated locations to examine for python installations")
```
Comment (thatch) [inline src/runez/__main__.py:52]: Help text isn't quite helpful enough.  Does the default imply that it reads the env var?  Can I specify any var?

File: src/runez/system.py [hunk 3]
```diff
@@ -387,27 +388,27 @@ def get_version(mod, default="0.0.0", fatal=False, logger=False):
     if name and isinstance(name, str):
         top_level = name.partition(".")[0] if isinstance(name, str) else name
```
Comment (thatch) [inline src/runez/system.py:389]: isinstance check is redundant

File: src/runez/system.py [hunk 4]
```diff
@@ -387,27 +388,27 @@ def get_version(mod, default="0.0.0", fatal=False, logger=False):
...
-        try:
-            from importlib import metadata  # requires py3.8+
-
-            version = metadata.version(top_level)
-            if version:
-                return version
-
-        except (ImportError, Exception) as e:
-            last_exception = e
+            except (ImportError, Exception) as e:
```
Comment (thatch) [inline src/runez/system.py:398]: ImportError inherits from Exception, why do you list it separately?

Comment (zsimic) [reply src/runez/system.py:398]: Ah indeed, fixed. Originally this was mixing both import errors (py version too low for `importlib.metadata.version` for example), and exceptions that may happen within the call itself.

File: src/runez/__main__.py (hunk 2 above)
Comment (zsimic) [reply src/runez/__main__.py:52]: It's just the plain string `PATH` (that represents the env var `PATH`), no other env var is supported.
Possibilities are explained here: https://github.com/codrsquad/runez/blob/main/src/runez/pyenv.py#L919-L923

File: src/runez/system.py (hunk 3 above)
Comment (zsimic) [reply src/runez/system.py:389]: Good catch! Fixed

**Pull request merged**